internal/ingress/annotations/satisfy/main_test.go

```diff
@@ -0,0 +1,133 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package satisfy
+
+import (
+	"testing"
+
+	api "k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/ingress-nginx/internal/ingress/annotations/parser"
+	"k8s.io/ingress-nginx/internal/ingress/resolver"
+)
+
+func buildIngress() *extensions.Ingress {
+	defaultBackend := extensions.IngressBackend{
+		ServiceName: "default-backend",
+		ServicePort: intstr.FromInt(80),
+	}
+
+	return &extensions.Ingress{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      "fake",
+			Namespace: api.NamespaceDefault,
+		},
+		Spec: extensions.IngressSpec{
+			Backend: &extensions.IngressBackend{
+				ServiceName: "default-backend",
+				ServicePort: intstr.FromInt(80),
+			},
+			Rules: []extensions.IngressRule{
+				{
+					Host: "fake.host.com",
+					IngressRuleValue: extensions.IngressRuleValue{
+						HTTP: &extensions.HTTPIngressRuleValue{
+							Paths: []extensions.HTTPIngressPath{
+								{
+									Path:    "/fake",
+									Backend: defaultBackend,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func TestIngressSatisfyConfigAny(t *testing.T) {
+	ing := buildIngress()
+
+	data := map[string]string{}
+
+	data[parser.GetAnnotationWithPrefix("satisfy")] = "any"
+	ing.SetAnnotations(data)
+
+	satisfyt, err := NewParser(&resolver.Mock{}).Parse(ing)
+	if err != nil {
+		t.Errorf("error parsing annotations: %v", err)
+	}
+
+	val, ok := satisfyt.(string)
+	if !ok {
+		t.Errorf("expected a string type but return %t", satisfyt)
+	}
+
+	if val != "any" {
+		t.Errorf("expected %v but returned %v", data[parser.GetAnnotationWithPrefix("satisfy")], val)
+	}
+}
+
+func TestIngressSatisfyConfigAll(t *testing.T) {
+	ing := buildIngress()
+
+	data := map[string]string{}
+
+	data[parser.GetAnnotationWithPrefix("satisfy")] = "all"
+	ing.SetAnnotations(data)
+
+	satisfyt, err := NewParser(&resolver.Mock{}).Parse(ing)
+	if err != nil {
+		t.Errorf("error parsing annotations: %v", err)
+	}
+
+	val, ok := satisfyt.(string)
+	if !ok {
+		t.Errorf("expected a string type but return %t", satisfyt)
+	}
+
+	if val != "all" {
+		t.Errorf("expected %v but returned %v", data[parser.GetAnnotationWithPrefix("satisfy")], val)
+	}
+}
+
+func TestIngressSatisfyConfigInvalid(t *testing.T) {
+	ing := buildIngress()
+
+	data := map[string]string{}
+
+	data[parser.GetAnnotationWithPrefix("satisfy")] = "invalid"
+	ing.SetAnnotations(data)
+
+	satisfyt, err := NewParser(&resolver.Mock{}).Parse(ing)
+	if err != nil {
+		t.Errorf("error parsing annotations: %v", err)
+	}
+
+	val, ok := satisfyt.(string)
+	if !ok {
+		t.Errorf("expected a string type but return %t", satisfyt)
+	}
+
+	if val != "all" {
+		t.Errorf("expected %v but returned %v", data[parser.GetAnnotationWithPrefix("satisfy")], val)
+	}
+}
```

Comment:
Similar comment as below, just assemble all inputs and expected output for all three test cases into a collection and iterate over them one by one and run the test case in single function instead of a function per case.

Reply:
Done